plugins/spi/SPIOutput.cpp

```diff
@@ -522,8 +522,13 @@ void SPIOutput::IndividualAPA102Control(const DmxBuffer &buffer) {
 
   // We always check out the entire string length, even if we only have data
   // for part of it
-  const uint16_t output_length = APA102_START_FRAME_BYTES +
+  // only add the APA102_START_FRAME_BYTES on the first port!!
+  if (m_output_number == 0) {
+    const uint16_t output_length = APA102_START_FRAME_BYTES +
       (m_pixel_count * APA102_SPI_BYTES_PER_PIXEL);
+  } else {
+    const uint16_t output_length = (m_pixel_count * APA102_SPI_BYTES_PER_PIXEL);
```

Comment:
Yes sorry, Iif you want to keep the const do:

const uint16_t output_length = (m_pixel_count * APA102_SPI_BYTES_PER_PIXEL) + 
                                                 (m_output_number ? 0 : APA102_START_FRAME_BYTES);